lib/config/validation.ts

```diff
@@ -205,10 +210,12 @@ export async function validateConfig(
         });
       }
       if (!optionTypes[key]) {
-        errors.push({
-          topic: 'Configuration Error',
-          message: `Invalid configuration option: ${currentPath}`,
-        });
+        if (!earlyCfg.allowExtraConfig) {
+          errors.push({
+            topic: 'Configuration Error',
+            message: `Invalid configuration option: ${currentPath}. Run the server with '--allow-extra-config' flag if you want to allow extra configurations`,
```

Comment:
Is there a reason to insist on/hint to the CLI parameter, and not the envvar, javascript or json option?